### DISCUSSION TITLE
Question: Dynamic types with module/namespace

### DISCUSSION THREAD

**Discussion by AlgoryxJosef** [eProsima/Fast-DDS#3061] [Q&A]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

I'm trying to achieve a message type e.g. `std_msgs::msg::dds_::Float32_`  where the Float32_ is a struct. 
I can do this when generating from an .idl using the `module` tag: 

```
module std_msgs
{
   module msg
   {
      module dds_
      {
         struct Float32_
         {
            float data;
         };
       }
     }
 }
``` 

Works great (and yes, it is for talking to a ROS2 node). 

I would like to get the same thing but for Dynamic Types.  I know the `module` keyword is not implemented in the .xml dynamic type schema. 

The closest I've come is to do: 

```
<types>
    <type>
        <struct name="std_msgs::msg::dds_::Float32_">
            <member name="data" type="float32"/>
        </struct>
    </type>
</types>
```

Which almost works; i get the data through to the ROS2 node, but I get an annoying warning "Error building type. The current descriptor isn't consistent".  

I've also tried using a struct builder directly from C++ with: 

```
struct_type_builder1->set_name("std_msgs::msg::dds_::Float32_");
```

Which does the same as the .xml situation above.

I saw another issue with a similar question, and the answer there was to, instead of using `::`, doing double underscores as `__`. I've tried this but the type simply does not match with what would be generated from the .idl at the top of this issue.

So my question is then; how can I get this to work? Struct in structs? Any other tricks?

Best regards,
Josef

### Current behavior

-

### Steps to reproduce

-

### Fast DDS version/commit

aa485fc

### Platform/Architecture

Windows 10 Visual Studio 2019

### Transport layer

Default configuration, UDPv4 & SHM

### Network traffic capture

[DYN_TYPES Error] Error building type. The current descriptor isn't consistent. -> Function eprosima::fastrtps::types::DynamicTypeBuilder::build

**Comment by JLBuenoLopez** [comment]
According to [Fast DDS Contributing guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md#contribution-guidelines) questions are to be asked in the corresponding Discussion forum. I am going to proceed and move this issue to its proper place.

**Reply by AlgoryxJosef** [reply]
Thank you [JLBuenoLopez-eProsima](https://github.com/JLBuenoLopez-eProsima)!
